module/VuFind/src/VuFind/View/Helper/Root/ProxyUrl.php

```diff
@@ -66,10 +78,66 @@ public function __construct($config = null)
      */
     public function __invoke($url)
     {
-        $usePrefix = !isset($this->config->EZproxy->prefixLinks)
-            || $this->config->EZproxy->prefixLinks;
+        $useWebService = $this->config->EZproxy->prefixLinksWebServiceUrl ?? false;
+        if ($useWebService) {
+            $usePrefix = $this->checkUrl($url) ?? $this->checkConfig();
```

Comment:
In theory some part of this could happen asynchronously so it did not hold up view rendering.  I think that is unnecessary because
- config.ini specifies the web service should be hosted closely so the call is trivially fast.
- responses are cached
- in many cases the URLs are part of templates which are already rendered asynchronously, such as a record tab or a combined bento box

Making this async would also add a lot of unnecessarily complexity.